### PR TITLE
trust4 module update

### PR DESCRIPTION
**Pull request by mapo9**
The previous version of the module was not sufficient to run trust4 on a large amount of different datasets. 

The updated version adds the possiblilty to give barcodes & umis to trust4 without requiring lots of changes in the code. 

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/trust4/main.nf
Comment (itrujnara) [inline modules/nf-core/trust4/main.nf:16]: Is there a reason the emit statements are so far out? You should align them to the end of the longest line if not

File: modules/nf-core/trust4/main.nf
Comment (itrujnara) [inline modules/nf-core/trust4/main.nf:50]: You have defined barcode to be an empty string, there is no reason to reassign the same value. Same goes for umi_read.

File: modules/nf-core/trust4/main.nf
Comment (itrujnara) [inline modules/nf-core/trust4/main.nf:43]: Modules cannot rely on the `meta` map containing any information other than `id` and `single_end` (cf. [modules docs](https://nf-co.re/docs/guidelines/components/modules#types-of-meta-fields)). You should refactor this to pass the information in another way. I know this might be trickier than usual (as the docs show no pre-processing on the variable), but there are solutions. My suggestion is to consider passing `barcode_read` and `umi_read`, and leave it to pipeline maintainers to correctly fetch the information form their `meta`. [This config](https://github.com/nf-core/modules/blob/71bc6bbffc7ad08e05c4d449ca3ba0f86b000125/subworkflows/nf-core/bam_split_by_region/nextflow.config) demonstrates how it might work inside a workflow.

**Review by itrujnara — CHANGES_REQUESTED** [review]
Hi, thanks for updating the module to match the needs of the community. I have put some comments. Most are rather cosmetic, but there is one important issue to address regarding `meta`.

**Comment by mapo9** [comment]
Hi @itrujnara,
thanks for the review and the pointers. 

I changed the functionality to require barcode & umi as individual inputs. Linked to a meta they can be null which allows the workflow for samples with and without barcodes/umis.

It would be great if you could have another look

**Review by itrujnara — APPROVED** [review]
Looks good now, thanks

**Pull request merged**
